src/arch/InputHandler/InputHandler_SDL.cpp

```diff
@@ -0,0 +1,974 @@
+    case SDLK_F12: return KEY_F12;
+    case SDLK_F13: return KEY_F13;
+    case SDLK_F14: return KEY_F14;
+    case SDLK_F15: return KEY_F15;
+    case SDLK_F16: return KEY_F16;
+
+    case SDLK_LCTRL: return KEY_LCTRL;
+    case SDLK_RCTRL: return KEY_RCTRL;
+    case SDLK_LSHIFT: return KEY_LSHIFT;
+    case SDLK_RSHIFT: return KEY_RSHIFT;
+    case SDLK_LALT: return KEY_LALT;
+    case SDLK_RALT: return KEY_RALT;
+    case SDLK_LGUI: return KEY_LMETA;
+    case SDLK_RGUI: return KEY_RMETA;
+    // Why are meta and super keys defined separately?
+    case SDLK_MENU: return KEY_MENU;
+    case SDLK_MODE: return KEY_FN;
+    case SDLK_NUMLOCKCLEAR: return KEY_NUMLOCK;
+    case SDLK_SCROLLLOCK: return KEY_SCRLLOCK;
+    case SDLK_CAPSLOCK: return KEY_CAPSLOCK;
+    case SDLK_PRINTSCREEN: return KEY_PRTSC;
+
+    case SDLK_UP: return KEY_UP;
+    case SDLK_DOWN: return KEY_DOWN;
+    case SDLK_LEFT: return KEY_LEFT;
+    case SDLK_RIGHT: return KEY_RIGHT;
+
+    case SDLK_INSERT: return KEY_INSERT;
+    case SDLK_HOME: return KEY_HOME;
+    case SDLK_END: return KEY_END;
+    case SDLK_PAGEUP: return KEY_PGUP;
+    case SDLK_PAGEDOWN: return KEY_PGDN;
+
+    case SDLK_KP_0: return KEY_KP_C0;
+    case SDLK_KP_1: return KEY_KP_C1;
+    case SDLK_KP_2: return KEY_KP_C2;
+    case SDLK_KP_3: return KEY_KP_C3;
+    case SDLK_KP_4: return KEY_KP_C4;
+    case SDLK_KP_5: return KEY_KP_C5;
+    case SDLK_KP_6: return KEY_KP_C6;
+    case SDLK_KP_7: return KEY_KP_C7;
+    case SDLK_KP_8: return KEY_KP_C8;
+    case SDLK_KP_9: return KEY_KP_C9;
+    case SDLK_KP_DIVIDE: return KEY_KP_SLASH;
+    case SDLK_KP_MULTIPLY: return KEY_KP_ASTERISK;
+    case SDLK_KP_MINUS: return KEY_KP_HYPHEN;
+    case SDLK_KP_PLUS: return KEY_KP_PLUS;
+    case SDLK_KP_PERIOD: return KEY_KP_PERIOD;
+    case SDLK_KP_EQUALS: return KEY_KP_EQUAL;
+    case SDLK_KP_ENTER: return KEY_KP_ENTER;
+    default: return DeviceButton_Invalid;
+    }
+}
+
+InputHandler_SDL::InputHandler_SDL()
+{
+    LOG->Info("[InputHandler_SDL] Using InputHandler_SDL");
+
+    m_pWindow = SDL_GL_GetCurrentWindow();
+    if (!m_pWindow)
+    {
+        LOG->Info("[InputHandler_SDL] Could not get window!");
+        return;
+    }
+
+    m_devicesChanged = false;
+    QueryJoysticks();
+}
+
+InputHandler_SDL::~InputHandler_SDL()
+{
+    for (auto pGamepad : m_pGamepads)
+    {
+        SDL_CloseGamepad(pGamepad);
+    }
+    for (auto pJoystick : m_pJoysticks)
+    {
+        SDL_CloseJoystick(pJoystick);
+    }
+    m_pGamepads.clear();
+    m_pJoysticks.clear();
+}
+
+void InputHandler_SDL::Update()
+{
+    if (!m_pWindow)
+    {
+        InputHandler::UpdateTimer();
+        return;
+    }
+
+    // Reset any wheel motion (we do not get notified of inaction)
+    ButtonPressed(DeviceInput(DEVICE_MOUSE, MOUSE_WHEELUP, 0));
+    ButtonPressed(DeviceInput(DEVICE_MOUSE, MOUSE_WHEELDOWN, 0));
+    INPUTFILTER->UpdateMouseWheel(0);
+
+    SDL_Event event;
+    while (SDL_PollEvent(&event))
```

Comment:
If we wanted to thread this we could run [SDL_PumpEvents](https://wiki.libsdl.org/SDL3/SDL_PumpEvents) on the main thread, and use [SDL_PeepEvents](https://wiki.libsdl.org/SDL3/SDL_PeepEvents) (which is thread safe according to the docs) but wanted to get input / testing in before I did any of that.